git/cmd.py

```diff
@@ -592,13 +602,45 @@ def execute(self, command,
         if as_process:
             return self.AutoInterrupt(proc, command)
 
+        def _kill_process(pid):
+            """ Callback method to kill a process. """
+            p = Popen(['ps', '--ppid', str(pid)], stdout=PIPE)
+            child_pids = []
+            for line in p.stdout:
+                if len(line.split()) > 0:
+                    local_pid = (line.split())[0]
+                    if local_pid.isdigit():
+                        child_pids.append(int(local_pid))
+            try:
+                os.kill(pid, SIGKILL)
+                for child_pid in child_pids:
+                    os.kill(child_pid, SIGKILL)
+                kill_check.set()    # tell the main routine that the process was killed
+            except OSError:
```

Comment:
If `os.kill` throws `OSError` if the process to be killed doesn't exist, I fear that after `os.kill(pid, SIGKILL)`, some of the child processes might already be down as `pid`. Right now, failure to kill the first child process, will prevent the others from being downed as well. Maybe something like that would help:

```python
for child_pid in child_pids:
    try:
        os.kill(child_pid, SIGKILL)
    except OSError:
        pass
```